dfm_tools/meshkernel_helpers.py

```diff
@@ -77,6 +77,43 @@ def meshkernel_delete_withgdf(mk:meshkernel.meshkernel.MeshKernel, coastlines_gd
                          delete_option=meshkernel.DeleteMeshOption(2), #ALL_COMPLETE_FACES/2: Delete all faces of which the complete face is inside the polygon
                          invert_deletion=False)
 
+def meshkernel_delete_withshp(mk:meshkernel.meshkernel.MeshKernel, coastlines_shp, min_area: float=0):
```

Comment:
you could consider adding a `crs` argument to convert the geodataframe to, in case of models/grids generated in local/cartesian coordinates instead of WGS84. An example can be found in a different function:
https://github.com/Deltares/dfm_tools/blob/dda0fb6fc251870ce51a2c9511bb02a6831875d0/dfm_tools/coastlines.py#L68-L69